crates/holochain/src/core/workflow/app_validation_workflow.rs

```diff
@@ -1,4 +1,97 @@
-//! The workflow and queue consumer for sys validation
+//! Holochain workflow to validate all incoming DHT operations with an
+//! app-defined validation function.
+//!
+//! Triggered by system validation, this workflow iterates over a list of
+//! [`DhtOp`]s to be validated, validates each op, updates its validation status
+//! in the database accordingly, and triggers op integration if necessary.
+//!
+//! ### Sequential validation
```

Comment:
```suggestion
//! ### Sequential validation (best effort)
```

Reply:
If best effort means "we're trying to do this sequentially but don't know if it is sequential", then that's not so. It is sequential for sure, but it would be an equal result if we validated all ops in parallel right now.

Reply:
Maybe I'm missing something but I don't think I agree. We sequentially validate based on the data that is available. If we're missing something and we don't know it's missing then we can't possibly work sequentially but we will attempt validation for what we do have in order... Maybe not a particularly helpful thing to raise in this section but that is how I'm thinking about this. Is that not right?

Reply:
That is right, but the ops are all validated in sequence no matter what the outcome is. All ops are validated one after the other, then that is repeated in the next workflow run.

Reply:
Ahh okay, yeah I see what you're saying. Yes that makes sense! I'll resolve this thread :)